src/App.jsx

```diff
@@ -1,10 +1,22 @@
-import { useState } from "react";
+import { useEffect, useState } from "react";
 import "./App.css";
 import AddTodoForm from "./AddTodoForm";
 import TodoList from "./TodoList";
 
+const useSemiPersistentState = () => {
+  const [todoList, setTodoList] = useState(
```

Comment:
```
const [todoList, setTodoList] = useState(() => {
  try {
    return JSON.parse(localStorage.getItem("savedTodoList")) || [];
  } catch (e) {
    console.error("Failed to parse savedTodoList:", e);
    return [];
  }
});
```
This might be better version:

- The code does not handle potential errors from JSON.parse. If the stored data is not valid JSON, it could cause an exception. Adding a try-catch block would make it more robust:

Reply:
Looks good! I just added it.